SteamKit2/SteamKit2/Steam/Handlers/SteamUser/SteamUser.cs

```diff
@@ -301,13 +313,19 @@ public void LogOn( LogOnDetails details )
 
             SteamID steamId = new SteamID( details.AccountID, details.AccountInstance, Client.ConnectedUniverse, EAccountType.Individual );
 
-            uint localIp = NetHelpers.GetIPAddress( this.Client.LocalIP );
+            if ( details.LoginID == null )
```

Comment:
Nit: Replace `== null` with `.HasValue` to be explicit about what is happening. (Roslyn does this when compiling anyway.)